store/generated_key/eigenda_v2/eigenda.go

```diff
@@ -0,0 +1,146 @@
+package eigendav2
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/avast/retry-go/v4"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/Layr-Labs/eigenda/api/clients/v2"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+type Config struct {
+	// address of service manager - used for loading chain state
+	ServiceManagerAddr string
+
+	MaxBlobSizeBytes uint64
+
+	// total duration time that client waits for blob to confirm
+	StatusQueryTimeout time.Duration
+
+	// number of times to retry eigenda blob dispersals
+	PutRetries uint
+}
+
+// Store does storage interactions and verifications for blobs with the
+// EigenDA V2 protocol.
+type Store struct {
+	cfg *Config
+	log log.Logger
+
+	disperser *clients.PayloadDisperser
+	retriever clients.PayloadRetriever
```

Comment:
nit: this probably should be `relayRetriever`, as there will now also be a `validatorRetriever`